Drag an Idea

ianmcnally opened the pull request:
__Description of Change(s) Introduced:__ 

Adds drag-and-drop to an Idea, re:  #336.

__Dependencies Introduced or Upgraded:__ (if applicable)

[babel-preset-stage-2](https://babeljs.io/docs/plugins/preset-stage-2/), can remove/break up dependency since it's not required, but it's become pretty handle and widely used for class-properties.

migrated babel-preset-es2015 to -env (recommended from babel)

- (Dependency) - (current version number)

__Description of Testing Applied:__ (if applicable)

Unit tests. Manually confirmed in latest Safari, Firefox, and Chrome.

__Relevant Screenshots/GIFs:__ (if applicable)

![drag and drop](https://user-images.githubusercontent.com/1494758/34573006-f983b0ac-f140-11e7-88cc-1679da0fa8ed.gif)


__Relevant github Issue:__ (if applicable)

 #336

vanderhoop commented:
hey @ianmcnally! awesome that you took this on! I'm billable on a client right now, so i'll have to do a more thorough review tonight, but on first look, I'm curious why you elected not to use `react-dnd`? My understanding is that the native HTML5 attributes and callbacks differ a bit from browser to browser, and react-dnd obscures those differences behind its API.

ianmcnally commented:
I'm always on that html-first train. Let me check on the compatibility stuff.

ianmcnally commented:
Tested in FF, Safari, Chrome. Couldn't connect to the web socket in Edge. Also looked at https://caniuse.com/#search=drag. Since this uses a small set of the features, I believe we're covered pretty well and don't see a need from react-dnd.

For browsers that don't support it, the user can simple use the existing edit feature.

vanderhoop commented on yarn.lock:
```diff
@@ -344,6 +344,22 @@ babel-generator@^6.26.0:
 
+babel-helper-bindify-decorators@^6.24.1:
+  version "6.24.1"
+  resolved "https://schoology.jfrog.io/schoology/api/npm/npm/babel-helper-bindify-decorators/-/babel-helper-bindify-decorators-6.24.1.tgz#14c19e5f142d7b47f19a52431e52b1ccbc40a330"
```
hey @ianmcnally! just tried to pull this and build it locally, and looks like you committed resolutions coming from schoology's local jfrog artifactory. Not a big deal, as I can always check out the changes, but I'm curious if you have a preferred solution for repos to ensure they're going to the public npm distributions (at least under the hood)? Lemmeknow!

vanderhoop replied:
honestly, i can just look this up myself 😄  but answer if you feel like it.

ianmcnally replied:
oh shoot, sorry, work computer's setup to point to an internal repo

vanderhoop commented:
hey @ianmcnally did a little diggin, and in `<IdeaReadOnlyContent />`, we have [all the ingredients](https://github.com/stride-nyc/remote_retro/blob/master/web/static/js/components/idea_read_only_content.jsx#L9) necessary for scoping this behavior to the appropriate stage and users

vanderhoop commented on package.json:
```diff
@@ -40,7 +40,9 @@
   "devDependencies": {
     "babel-core": "^6.17.0",
     "babel-loader": "^7.1.2",
+    "babel-preset-env": "^1.6.1",
```
thanks for doing this ❤️  i've been blind to this wart for too long.

ianmcnally commented:
@vanderhoop thanks for providing such clear feedback and next steps. moved the dragging from idea to idea-readonly, and added the logic checks you suggested. holla, holla. 3861bfe
